### ISSUE TITLE
automatically generate document id

### ISSUE THREAD

**Issue by tommyliu1114** [elastic/go-elasticsearch#79]
using this,i can not add a document without document unique(_id);
 i check the package "api", i found all the create method based on http method "put".
so,do you  have a plan to support automatically generate document id with http method "Post"?

**Comment by philkra** [comment]
Hi @tommyliu1114 , can you please provide the code snippet you are struggling with? Are you asking that the client should generate the document `id` when indexing a document?

**Comment by karmi** [comment]
@tommyliu1114 , you mean that the `id` parameter is required in the _Create()_ API method (https://godoc.org/github.com/elastic/go-elasticsearch/esapi#Create)? This is indeed incorrect — the code is generated, I'll have a look into fixing the generator.

In the meantime, if you'd use the [`CreateRequest`](https://github.com/elastic/go-elasticsearch/blob/master/esapi/api.create.go#L36) struct directly, it should work by simply setting ID to an empty string.

**Comment by tommyliu1114** [comment]
hey,karmi,Thank you very much。 i found the solution , you have offered method in api.index.go ，so i use esapi.IndexRequest.
Nice Day! Nice work!

**Comment by carstenpatzke** [comment]
Can someone reopen this issue?
It seems like Create still requires an id.

**Comment by karmi** [comment]
@carstenpatzke, I've looked into it a bit, and the ID parameter is required for the Create API in language clients. The [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html) is not 100% clear on the distinction between the Index and Create APIs, but the REST API [specification](https://github.com/elastic/elasticsearch/blob/1cf789e5fcfb0f364f665bfaac021e24a4c2f556/rest-api-spec/src/main/resources/rest-api-spec/api/create.json#L11) is clear on that.

**Comment by jaxsonbloggs** [comment]
I'm trying to insert a new record and have an ID automatically generated but I'm finding that if I set the DocumentID to an empty string it will create a record on the first request then update it on any following requests. I've attempted this using IndexRequest and CreateRequest with the Go API. I've also attempted this creating a  _create POST API request via Postman. Has anyone found a solution?
